Gems/EMotionFX/Code/EMotionFX/Source/Motion.cpp

```diff
@@ -335,4 +336,14 @@ namespace EMotionFX
         }
         m_motionData = motionData;
     }
+
+    void Motion::SetRootMotionExtractionData(const AZStd::shared_ptr<RootMotionExtractionData>& data)
+    {
+        m_rootMotionExtractionData = data;
+    }
```

Comment:
update: Again to show this is a sink parameter (taking ownership) passing by value is usually clearer

```c++
void Motion::SetRootMotionExtractionData(AZStd::shared_ptr<RootMotionExtractionData> data)
{
    m_rootMotionExtractionData = AZStd::move(data);
}
```

Reply:
https://dzone.com/articles/c-sink-parameter-passing found a great article on sink param. Good suggestion!

Reply:
Oh awesome thanks! 🙂